logging/slog_logger.go

```diff
@@ -1,75 +1,119 @@
 package logging
 
 import (
+	"context"
 	"fmt"
+	"io"
 	"log/slog"
 	"os"
+	"runtime"
+	"time"
+)
+
+type SlogHandlerType string
+
+const (
+	SLogHandlerTypeJson     SlogHandlerType = "json" // prints debug and above
+	SLogHandlerTypeJsonText SlogHandlerType = "text" // prints info and above
 )
 
 type SLogger struct {
-	logger *slog.Logger
+	*slog.Logger
 }
 
 var _ Logger = (*SLogger)(nil)
 
-func NewSlogLogger(env LogLevel) Logger {
-	if env == Production {
-		logHandler := slog.NewJSONHandler(os.Stdout, nil).WithAttrs([]slog.Attr{slog.String("env", string(env))})
-		logger := slog.New(logHandler)
-		return &SLogger{
-			logger: logger,
-		}
-	} else if env == Development {
-		// Even though code is same for both environments, we are keeping it separate to show that we can have different
-		// implementations for different environments. For example, we can have log levels set differently
-		logHandler := slog.NewJSONHandler(os.Stdout, nil).WithAttrs([]slog.Attr{slog.String("env", string(env))})
-		logger := slog.New(logHandler)
-		return &SLogger{
-			logger: logger,
-		}
-	} else {
-		panic(fmt.Sprintf("Unknown environment. Expected %s or %s. Received %s.", Development, Production, env))
+// NewSlogTextLogger creates a new SLogger with a text handler
+//
+//	outputWriter is the writer to write the logs to (typically os.Stdout,
+//	  but can also use a io.MultiWriter(os.Stdout, file) to write to multiple outputs)
+//	logLevel is the minimum level to log
+//	logSource if true, adds source information to the log
+func NewSlogTextLogger(outputWriter io.Writer, logLevel slog.Level, logSource bool) *SLogger {
```

Comment:
I feel this should expect a log config which should have: env, log level etc - whatever is needed.

Reply:
Shall we take the entire `slog.HandlerOptions` as input then?
So `NewSlogTextLogger(outputWriter io.Writer, handerOptions slog.HandlerOptions)`

Reply:
I think that makes sense or something like
```
type SLoggerConfig struct {
 slog.HandlerOptions
 OutputWriter io.Writer
}
```

@shrimalmadhur wdyt?

Reply:
See [2bfeb3f](https://github.com/Layr-Labs/eigensdk-go/pull/127/commits/2bfeb3f7f7f662167474e29c4f7a243ffe0aa3fa)